test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/End2EndTests.cs

```diff
@@ -32,47 +32,116 @@
-            OpenIdConnectConfiguration configuration = OpenIdConnectConfigurationRetriever.GetAsync(OpenIdConfigData.OpenIdConnectMetadataFileEnd2End, new FileDocumentRetriever(), CancellationToken.None).Result;
-            JwtSecurityTokenHandler tokenHandler = new JwtSecurityTokenHandler();
-            JwtSecurityToken jwtToken = 
-                tokenHandler.CreateJwtSecurityToken(
-                    configuration.Issuer,
-                    Default.Audience,
-                    ClaimSets.DefaultClaimsIdentity,
-                    DateTime.UtcNow,
-                    DateTime.UtcNow + TimeSpan.FromHours(1),
-                    DateTime.UtcNow + TimeSpan.FromHours(1),
-                    rsaSigningCredentials);
+            try
+            {
+                OpenIdConnectConfiguration configuration = OpenIdConnectConfigurationRetriever.GetAsync(theoryData.OpenIdConnectMetadataFileName, new FileDocumentRetriever(), CancellationToken.None).Result;
+                JwtSecurityTokenHandler tokenHandler = new JwtSecurityTokenHandler();
+                JwtSecurityToken jwtToken =
+                    tokenHandler.CreateJwtSecurityToken(
+                        configuration.Issuer,
+                        Default.Audience,
+                        ClaimSets.DefaultClaimsIdentity,
+                        DateTime.UtcNow,
+                        DateTime.UtcNow + TimeSpan.FromHours(1),
+                        DateTime.UtcNow + TimeSpan.FromHours(1),
+                        theoryData.SigningCredentials);
 
-            tokenHandler.WriteToken(jwtToken);
+                tokenHandler.WriteToken(jwtToken);
 
-            TokenValidationParameters validationParameters =
-                new TokenValidationParameters
-                {
-                    IssuerSigningKeys = configuration.SigningKeys,
-                    ValidAudience = Default.Audience,
-                    ValidIssuer = configuration.Issuer,
-                };
+                TokenValidationParameters validationParameters =
+                    new TokenValidationParameters
+                    {
+                        IssuerSigningKeys = configuration.SigningKeys,
+                        ValidAudience = Default.Audience,
+                        ValidIssuer = configuration.Issuer,
+                    };
+
+                tokenHandler.ValidateToken(jwtToken.RawData, validationParameters, out SecurityToken securityToken);
 
-            SecurityToken securityToken = null;
-            tokenHandler.ValidateToken(jwtToken.RawData, validationParameters, out securityToken);
+                theoryData.ExpectedException.ProcessNoException(context);
+            }
+            catch (Exception ex)
```

Comment:
very generic. shouldn't it be a more specific exception?  (related to the product question)

Reply:
This is how our testing framework works. We'll catch any exception, but the exception will be compared with an expected exception specified in theory data.
e.g. ` ExpectedException = ExpectedException.SecurityTokenSignatureKeyNotFoundException()`, for this scenario.